src/components/Verse/VerseText.tsx

```diff
@@ -28,7 +28,7 @@ const VerseText = ({ words }: VerseTextProps) => {
       <StyledVerseText styles={quranReaderStyles}>
         {words?.map((word) => (
           <QuranWord
-            key={[word.position, word.code, word.lineNum].join('-')}
+            key={word.location || word.id}
```

Comment:
some words don't have location data, need to fix the data. FYI: location is combination of chapter:verse:wordPosition and it'll always be unique for each word.

Reply:
Word location is combination of ayah key and word position within ayah.  

Location of first word of 1:1 would be `1:1:1`. Can't use id because,  it's not uniq. At least for first surah. We'll have ayah with id=1 and word with id=1